processing/src/main/java/org/apache/druid/timeline/DataSegmentChange.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.timeline;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.apache.druid.java.util.common.StringUtils;
+
+public class DataSegmentChange
+{
+  private final SegmentWithOvershadowedStatus segmentWithOvershadowedStatus;
+  private final boolean load;
```

Comment:
I think `load` is false only when change type is `SEGMENT_REMOVED`, maybe we can get rid of the `load` field if it doesn't add any new info. 

For convenience, you may still have an `isLoad` (maybe better named `isSegmentRemoved`) method but it should not be serializable (use @JsonIgnore if needed).